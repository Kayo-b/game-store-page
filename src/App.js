import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <div>
        <h1>Hello App</h1>
      </div>
    </div>
  );
}

export default App;
