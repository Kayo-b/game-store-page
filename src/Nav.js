import React from 'react';
import { Link } from 'react-router-dom'

const Nav = () => {
    return(
        <nav>
            <h3>nav</h3>
             <ul className="nav-links">
                <Link to='/about'>
                    <li>About</li>    
                </Link>
                <Link to='/shop'>
                    <li>Shop</li> 
                </Link>

             </ul>
        </nav>
    )
}

export default Nav