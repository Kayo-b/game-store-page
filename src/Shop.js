import React, { useState, useEffect }from 'react';
import { Link } from 'react-router-dom';

const Shop = () => {
    const [nationalities, setNationalities] = useState([]);
    const [message, setMessage] = useState('');
    const [personName, setPersonName] = useState('');
    const [items, setItems] = useState([])
    const [apiList, setApiList] = useState([
                    'https://www.cheapshark.com/api/1.0/deals?storeID=1', 
                    'https://www.cheapshark.com/api/1.0/deals?storeID=6'])

    async function fetchItems() {
        let allGamesArray = [];
        for(let x = 0; x < apiList.length; x++) {
            const data = await(await fetch(apiList[x])).json();
            allGamesArray = allGamesArray.concat(data)
        }

        // const data = await(await fetch('https://www.cheapshark.com/api/1.0/deals?storeID=1')).json();
        // const amount = data.length
        // const gamesNumber = amount ? data.country : [];
        

        // const otherStore = await fetch('https://www.cheapshark.com/api/1.0/deals?storeID=6')
        // .then(res => res.json())

        // setItems(otherStore.concat(otherStore))

        // let allGamesArray = data.concat(otherStore)
        setItems(allGamesArray)

        // var dataMain = Promise.all([fetch('https://www.cheapshark.com/api/1.0/deals?storeID=1&upperPrice=15'), fetch('https://www.cheapshark.com/api/1.0/deals?storeID=6&upperPrice=15')])
        // .then(res => Promise.all(res.map(res => res.json())))
        // .then(data => data)
        // console.log(dataMain)
     
        console.log(items)
        console.log(items.length)
        const message = allGamesArray.length > 0 ? `${allGamesArray.length} deals found` : 'No deals found';
        setMessage(message);

    // } catch (err) {
    //     console.log(`err: ${err.message}`);
    //     setNationalities([]);
    //     setMessage('Could not fetch deals, try again later.');
    // }
  }

  useEffect(() => {
    async function handleSubmit(){
        
        await fetchItems();
      }
      handleSubmit()
  },[])  


  return (
    <div className="shop">
        <div className="message">{message}</div>
        {console.log(items)}
        {items?.map(item => { 
           let gameName = `${item.metacriticLink}`
            console.log(gameName)
            gameName = gameName.slice(9)
            return <div key={item.title} className="gameDiv">

                <a href={`https://www.cheapshark.com/redirect?dealID={${item.dealID}}`}>
                    <h5 className="game-title">{item.title}</h5>
                </a>

                <img className="thumbnail" src={item.thumb} alt="thumbnail"></img>

                 <Link to={`/shop/${gameName}`} state={{name: gameName}}>
                    <div className="details">Info</div>
                </Link> 
        {/* <ItemsList gameName={gameName}/> */}
    </div>
})}
           
          </div>   
  );

}
// {Array.isArray(items) && items.map(
//     item => {
//       var gameName = `${item.metacriticLink}`
      
//       gameName = gameName.slice(9)
//       const flagUrl = item.thumb;
//       const altText = `${item.title} deal`;
//       return <div key={item.title}><h3>{item.title}</h3> 
//       <img src={flagUrl} alt={altText} style={{
//         border: "1px solid black"
//       }} />
      
//        <Link to={`/shop/${gameName}`} state={{name: gameName}}><div className="details">Info</div> </Link>
//       </div>
//     }
//   )}

export function ItemsList({ gameName }) { 
    return <Link to={`/shop/${gameName}`} state={{name: gameName}}><div className="details">Info</div></Link> }
//     return items?.map(item => { 
//     let gameName = `${item.metacriticLink}`
//     console.log(gameName)
//     gameName = gameName.slice(9)
//     return <div className="gameDiv">

//         <a href={`https://www.cheapshark.com/redirect?dealID={${item.dealID}}`}>
//             <h5 className="game-title">{item.title}</h5>
//         </a>

//         <img className="thumbnail" src={item.thumb} alt="thumbnail"></img>
        
      
//         <div><Link to={`/shop/${gameName}`} state={{name: gameName}}><div className="details">Info</div></Link> </div>  
       

        
//     </div>
// })}

// const Shop = () => {
    
//     useEffect(() => {
        
//         fetchItems();
//     }, [])

//     const [items, setItems] = useState([])

//     const fetchItems = async () => {
//         try {

//             const steamData = await fetch('https://www.cheapshark.com/api/1.0/deals?storeID=1&upperPrice=15')
//             // .then(res=>res.json())
//             // .then(print=>console.log(print))
//             const steamDeals = await steamData.json()
//             setItems(steamDeals)

//         } catch (err) {
//             console.log(`err: ${err.message}`);
//             setItems([])
//         }


//         // const requestOptions = {
//         //     method: 'POST',

//         // }
//         // const requestHeaderIGDB = {
//         //     method: 'POST',
//         //     headers: {
//         //         "Client-ID":"emjxtjdm4r6zsam311xpzv6ne7renk",
//         //         "Authorization": "Bearer tgwq7cyn3zbcvcn1tu6sbgpbpndj5w "
//         //     },
//         //     Body: "fields *;"
//         // }
//         // fetch("https://id.twitch.tv/oauth2/token?client_id=emjxtjdm4r6zsam311xpzv6ne7renk&client_secret=z2is0jtg3m5shv0m93g5by3xfar0o5&grant_type=client_credentials", requestOptions)
//         // .then(res=>res.json())
//         // .then(json=>console.log(json))
//         // fetch("https://api.igdb.com/v4/games",requestHeaderIGDB)
//         // .then(res=>res.json())
//         // // .then(json=>console.log(json))

//         // const otherStore = await fetch('https://www.cheapshark.com/api/1.0/deals?storeID=6&upperPrice=15')
//         // .then(res => res.json())

//         // setItems(steamDeals.concat(otherStore))
//         // setItems(steamDeals)

//         // const rawg = await fetch("https://api.rawg.io/api/games/corto-maltese-secrets-of-venice?key=61dfdbccb6ec4b6088ca3f707f739cdb", {method:'GET'})
//         // const games = await rawg.json();
//         // console.log(games)
//     } 
//     async function handleSubmit(e) {
//         e.preventDefault();
//         await fetchItems();
//     }

//     return(
//         <div className="shop">
//             <ItemsList items={items}/>
               
//         </div>
//     )
// }

export default Shop
