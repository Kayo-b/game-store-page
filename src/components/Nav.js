import React from 'react';
import { Link } from 'react-router-dom'

const Nav = ({shopCart}) => {
    return(
        <nav>
            <h3>nav</h3>
             <ul className="nav-links">
                <Link to='/'>
                    <li>Home</li>    
                </Link>
                <Link to='/deals'>
                    <li>Deals</li> 
                </Link>
                <Link to='/about'>
                    <li>About</li>    
                </Link>

                <Link to='/cart'>
                    <li>{shopCart.length} Cart</li>    
                </Link>

             </ul>
        </nav>
    )
}

export default Nav