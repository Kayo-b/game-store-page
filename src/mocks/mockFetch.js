const gamesArray = [
  {
  internalName: "SIDMEIERSCIVILIZATIONVI",
  title: "Sid Meiers Civilization VI",
  metacriticLink: "sid-meiers-civ"
  },
  {
    internalName: "2SIDMEIERSCIVILIZATIONVI",
    title: "2 Sid Meiers Civilization VI",
    metacriticLink: "2-sid-meiers-civ"
    }
]

export default async function mockFetch(url) {
  if(url.startsWith('https://www.cheapshark.com/api/')) {
    return {
      ok: true,
      status: 200,
      json: async () => gamesArray,
    };
  }


  throw new Error(`Unhandled request: ${url}`);        
}