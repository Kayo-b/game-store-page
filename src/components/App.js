import logo from '/home/kayo-b/the_odin_project/game-store-page/src/logo.svg'
import '/home/kayo-b/the_odin_project/game-store-page/src/App.css';


function App() {
  return (
    <div className="App">
      <div>
        <h1>Hello App</h1>
      </div>
    </div>
  );
}

export default App;
