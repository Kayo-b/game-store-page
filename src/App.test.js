import React from 'react';
import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom'
import userEvent from "@testing-library/user-event";
import mockFetch from './mocks/mockFetch'
import App from './App';
import Nav from './Nav';
import Shop from './Shop';
import { ItemsList } from './Shop';


// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
let windowFetchSpy;
beforeEach(() => {
  windowFetchSpy = jest.spyOn(window, 'fetch').mockImplementation(mockFetch)

});

afterEach(() => {
  jest.restoreAllMocks();
})


describe("App component", () => {
  it("renders correct heading", () => {
    render(<App />);
    const appHeader = screen.getByRole("heading");
    expect(appHeader.textContent).toMatch(/Hello App/i);
  });
});

describe("Nav component", () => {
  it("renders  nav title, About and Shop", () => {
    render(<Nav/>, {wrapper:BrowserRouter});
    const navHeader = screen.getByRole("heading");
    const listItems = screen.getAllByRole("listitem")

    expect(navHeader.textContent).toMatch(/nav/i);
    expect(listItems[0].textContent).toMatch(/about/i);
    expect(listItems[1].textContent).toMatch(/shop/i);
    // expect(shop.textContent).toMatch(/shop/i);
  });
  
});

describe("Shop component", () => {
  

    it("renders correct link", () => {
      const { container } = render(<Shop />)
      expect(container).toMatchSnapshot();
    });

    it("fetch was called", async () => {
        render(<Shop />);
        
        // const {getByRole} = render(<Shop />)
        // expect(getByRole("heading").textContent).toMatch(/header test/i)
        expect(windowFetchSpy).toHaveBeenCalled();
        
        
        // const gameShark = screen.getByText(/info/i)
        // expect(gameShark).toBeInTheDocument()
        // userEvent.click(gameShark)
    })


    // it("renders game name", async () => {
    //   render(<Shop />);
    // //   render(<ItemsList/>)
    // //   function sleep(period) {
    // //     return new Promise (resolve => setTimeout(resolve, period))
    // // }
    // // await act(async() => {
    // //     await sleep(2000)
    // // })
    // // await new Promise(process.nextTick);
    //   const title = screen.getByText(/teste/i)
    //   expect(title).toBeInTheDocument()
    // })

    test('should get nationalities for a name', async() => {
      render(<BrowserRouter><Shop /></BrowserRouter>);
    
      //simulate filling up the textbox
      // const personNameInput = screen.getByRole('textbox');
      // fireEvent.change(personNameInput, {target: {value: 'hn'}})
      // expect(personNameInput.value).toBe('hn');
    
      // //click the button
      //  const getNationalitiesBtn = screen.getByRole('button', { name: 'Get gud' });
      //  expect(getNationalitiesBtn).not.toBeDisabled();
      //  userEvent.click(getNationalitiesBtn);
    
       //verify percent and flag images are displayed
  
       expect(await screen.findByText('4 deals found')).toBeVisible();
       
    
       expect(windowFetchSpy).toHaveBeenCalled();
       expect(windowFetchSpy).toHaveBeenCalledWith('https://www.cheapshark.com/api/1.0/deals?storeID=1');
       expect(screen.getAllByText('Sid Meiers Civilization VI')[0]).toBeVisible();
       expect(screen.getAllByText('2 Sid Meiers Civilization VI')[1]).toBeVisible();
      //  expect(screen.getByText('IE - 4.21%')).toBeVisible();
      
       const flagImages = screen.getAllByRole('img');
       expect(flagImages).toHaveLength(4);
      //  expect(flagImages[0]).toHaveAccessibleName('US flag');
      //  expect(flagImages[1]).toHaveAccessibleName('IM flag');
      //  expect(flagImages[2]).toHaveAccessibleName('IE flag');
    });
  }


  
  );