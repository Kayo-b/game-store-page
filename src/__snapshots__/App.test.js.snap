// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component renders correct link 1`] = `
<div>
  <div
    class="shop"
  >
    <div
      class="message"
    />
  </div>
</div>
`;
